apps/authentication-service/src/shared/tokenUtils.test.ts

```diff
@@ -0,0 +1,144 @@
+import { getTestToken } from "@/test/testData/users";
+import fs from "fs";
+
+import {
+  getCachedToken,
+  getIsTokenExpired,
+  getLocalToken,
+  logToken,
+  setCachedToken,
+  setLocalToken,
+} from "@/shared/tokenUtils";
+
+let tokenBackup: string;
+const validTestToken: string = getTestToken();
+const expiredTestToken: string = getTestToken(true);
+
+describe("tokenUtils", () => {
+  beforeAll(() => {
+    tokenBackup = getCachedToken() || "";
+  });
+
+  afterAll(() => {
+    if (tokenBackup) {
+      setCachedToken(tokenBackup);
+    }
+    jest.clearAllMocks();
+  });
+
+  describe("setCachedToken", () => {
+    it("sets token", () => {
+      jest.spyOn(fs, "existsSync").mockReturnValue(true);
+      jest.spyOn(fs, "writeFileSync").mockImplementation(() => {});
+      jest.spyOn(fs, "readFileSync").mockReturnValue(validTestToken);
+      setCachedToken(validTestToken);
+      expect(getCachedToken()).toBe(validTestToken);
+    });
+
+    it("throws an error when unable to cache token", () => {
+      jest.spyOn(fs, "writeFileSync").mockImplementation(() => {
+        throw new Error();
+      });
+
+      const result = setCachedToken(validTestToken);
+      expect(result).toBe(false);
+    });
+  });
+
+  describe("getCachedToken", () => {
+    it("returns undefined", () => {
+      jest.spyOn(fs, "existsSync").mockReturnValue(false);
+      expect(getCachedToken()).toBeUndefined();
+    });
+
+    it("returns token", () => {
+      jest.spyOn(fs, "existsSync").mockReturnValue(true);
+      jest.spyOn(fs, "readFileSync").mockReturnValue(validTestToken);
+      expect(getCachedToken()).toBe(validTestToken);
+    });
+  });
+
+  describe("getLocalToken", () => {
+    it("returns undefined", () => {
+      setLocalToken(undefined);
+      expect(getLocalToken()).toBeUndefined();
+    });
+
+    it("returns token", () => {
+      setLocalToken(validTestToken);
+      expect(getLocalToken()).toBe(validTestToken);
+    });
+  });
+
+  describe("setLocalToken", () => {
+    it("sets token", () => {
+      setLocalToken(validTestToken);
+      expect(getLocalToken()).toBe(validTestToken);
+    });
+  });
+
+  describe("getIsTokenExpired", () => {
```

Comment:
The stuff in this describe is great. Generating test tokens gives us a lot of confidence with these tests.